naucse_hooks.py

```diff
@@ -27,6 +32,25 @@ def get_latest_naucse() -> Path:
     return path
 
 
+def get_last_commit_in_branch(repo, branch):
+    parsed = giturlparse.parse(repo)
+
+    if not parsed.valid:
+        return None
+
+    if not parsed.github:
+        return None
+
+    url = f"https://api.github.com/repos/{parsed.owner}/{parsed.repo}/commits/{branch}"
+
+    try:
+        response = requests.get(url)
```

Comment:
Travis looks OK, but a random stranger can still trigger an unlimited number of GitHub API requests.
I don't think it's a critical issue.

Reply:
I could cache the results of these functions for some reasonable time (like a second or smth), so the number of requests to GitHub API is limited at least in some way.